src/common/sp-strings.adb

```diff
@@ -13,34 +13,40 @@
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
 -------------------------------------------------------------------------------
-with Ada.Strings.Fixed;
+
 with Ada.Characters.Latin_1;
+with Ada.Strings.Fixed;
+with Ada.Strings.Unbounded;
 
 package body SP.Strings is
+
+    package ASU renames Ada.Strings.Unbounded;
+
     function Zip (Left : SP.Strings.String_Vectors.Vector; Right : SP.Strings.String_Vectors.Vector)
-        return Ada.Strings.Unbounded.Unbounded_String
+        return String
     is
         use Ada.Strings.Unbounded;
         use SP.Strings.String_Vectors;
         L : Natural := 1;
         R : Natural := 1;
+        Result : Ada.Strings.Unbounded.Unbounded_String;
     begin
-        return Result : Ada.Strings.Unbounded.Unbounded_String do
-            while L <= Natural (Length (Left)) or else R <= Natural (Length (Right)) loop
-                if L <= Natural (Length (Left)) then
-                    Append (Result, Left (L));
-                    L := L + 1;
-                end if;
+      while L <= Natural (Length (Left)) or else R <= Natural (Length (Right)) loop
+         if L <= Natural (Length (Left)) then
+            Append (Result, Left (L));
+            L := L + 1;
+         end if;
 
-                if R <= Natural (Length (Right)) then
-                    Append (Result, Right (R));
-                    R := R + 1;
-                end if;
-            end loop;
-        end return;
+         if R <= Natural (Length (Right)) then
+            Append (Result, Right (R));
+            R := R + 1;
+         end if;
+      end loop;
+
+      return To_String (Result);
```

Comment:
This looks like it results in an additional copy to the stack on every call to this function.